ggml.c

```diff
@@ -7799,7 +7799,9 @@ static void ggml_compute_forward_mul_mat_q_f32(
     //   compute by src0 rows
 
 #if defined(GGML_USE_CUBLAS)
-    if (ggml_cuda_can_mul_mat(src0, src1, dst)) {
+    if (ggml_cuda_can_mul_mat(src0, src1, dst) ||
+        ggml_cuda_can_dequantize_mul_mat_vec(src0, src1, dst) ||
+        src0->backend == GGML_BACKEND_CUDA) {
```

Comment:
Is there any reason to put this here instead of leaving all the checks in `ggml_cuda_can_mul_mat`?

Reply:
I was initially thinking it would be better this way: to have a separate method for dequantize mul mat since you can then also use that method for the logic inside `ggml_cuda_mul_mat_q_f32`. I was thinking I would just push it and see what other people have to say since it turned out kind of convoluted.

Reply:
I think that it is better to keep the complexity in `ggml_cuda_can_mul_mat` than to spread it all around, but I am not entirely sure if you have any reason to split the check in two parts.